domain/internal/model/hint_test.go

```diff
@@ -0,0 +1,49 @@
+package model
+
+import (
+	"encoding/json"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	cwd, _   = os.Getwd()
+	testdata = path.Join(cwd, "testdata")
+)
+
+func Load(name string) *Board {
+	jsonData, _ := os.ReadFile(path.Join(testdata, name))
+
+	var board Board
+	json.Unmarshal(jsonData, &board)
+	return &board
+}
+
+func TestHint(t *testing.T) {
```

Comment:
I'm not aware of test practices in go language but I'd prefer to have separates tests. I don't care having duplications in tests, all I want is a clear, easy to read test. I'd imagine something like : 
```go
t.Run("GetPlaceTileHint should return the right hint", func(t *testing.T) {
	board := Load("place-tile-hint");

	updatedBoard, hint := board.GetPlaceTileHint()

	assert.NotEqual(t, board, updatedBoard)
	assert.Equal(t, test.hint, hint)
})

t.Run("GetPlaceTileHint should return nil if no move lead to the goal", func(t *testing.T) {
	board := Load("place-tile-no-hint");

	updatedBoard, hint := board.GetPlaceTileHint()

        assert.Equal(t, board, updatedBoard)
	assert.Equal(t, test.hint, nil)
})
```

Reply:
It's not contradictory, you can use test data AND split the tests

Reply:
> It's not contradictory, you can use test data AND split the tests

I splitted the tests as it is more clear